Controller DENON MC7000 files updated

holopansel opened the pull request:
changes include play/pause button with softstart and break, rate selectable by knob,
pitchplay included,
sampler updated to up to 32 samplers, selectable with variable sampler qty
velocity sampler included

holopansel committed:
play, pause with softstart and break;
pitch play;
velocity sampler;
sampler quantity up to 32;

holopansel committed:
play, pause with softstart and break;
pitch play included;
velocity sampler included;
sampler changed to maximum quantity 32;

daschuer commented:
I am sorry, there are eslint issues, see: https://github.com/mixxxdj/mixxx/runs/5824817384?check_suite_focus=true
There have been probably stacked up before we added eslint as a CI stage. But can you take care of them? 
You can also install it locally: https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking

holopansel committed:
update according to code style issue log

holopansel committed:
Merge branch 'mixxxdj:main' into main

holopansel committed:
update after code check

holopansel commented:
Thanks Daniel,I updated the file, how do I proceed now?Do I need to trigger a new pull request?Or is there a way to continue with the existing one?Sorry for the stupid question, first time to work with git.Best regards,John Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Daniel Schürmann ***@***.***> Datum: 05.04.22  07:52  (GMT+01:00) An: mixxxdj/mixxx ***@***.***> Cc: holopansel ***@***.***>, Author ***@***.***> Betreff: Re: [mixxxdj/mixxx] Controller DENON MC7000 files updated (PR #4712) 
I am sorry, there are eslint issues, see: https://github.com/mixxxdj/mixxx/runs/5824817384?check_suite_focus=true
There have been probably stacked up before we added eslint as a CI stage. But can you take care of them?
You can also install it locally: https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

ronso0 commented on res/controllers/Denon-MC7000-scripts.js:
```diff
@@ -621,9 +624,9 @@ MC7000.padModePitch = function(channel, control, value, status, group) {
     MC7000.PADModePitch[deckNumber] = true;
 
-    // switch on initial PAD illumination = hotcue for pitch or if HotcueSelectedGroup selected = pad mode pitchoff color
+    // switch on initial PAD illumination = hotcue for pitch or if MC7000.HotcueSelectedGroup selected = pad mode pitchoff color
```
```suggestion
    // switch on initial PAD illumination = hotcue for pitch or if
    // MC7000.HotcueSelectedGroup selected = pad mode pitchoff color
```

holopansel replied:
Done that, will update the comment.

ronso0 commented:
> I updated the file, how do I proceed now?

Since this is your first contribution a team member needs to trigger a CI run manually after every change. I just did that.

daschuer commented:
I have copied the requested changes from eslint to https://github.com/holopansel/mixxx/pull/1
Merge it if you like, but I strongly recommend ti install pre-commit linked above.

holopansel commented:
Regarding the code style issues, if I change all var to const or let like suggested, the file does no longer work with mixxx. Apart from that I do not like to declare a const which is subject to change during execution.

> I have copied the requested changes from eslint to [holopansel#1](https://github.com/holopansel/mixxx/pull/1) Merge it if you like, but I strongly recommend ti install pre-commit linked above.

Already installed pre-commit. If I replace the var with let like suggested in the code style issues, mixxx reports that the file does not work correctly, expecting more semicolons. Adding these does not do any change. So as a conclusion I cannot make the file work with let instead of var. 
Any suggestions for that would be highly appreciated.

Swiftb0y commented:
> Regarding the code style issues, if I change all var to const or let like suggested, the file does no longer work with mixxx.

The issue is that you are working on a mapping that is made for Mixxx 2.4 (alpha), but you are running 2.3 (stable) on your machine. So you either have the choice of installing mixxx 2.4 (thereby testing the unstable alpha version) or you checkout the 2.3 branch in git, and work with that version of the mapping (which also still has the appropriate code style linter configuration).

holopansel commented:
Thank you, understood. So there are two things that I need to do now, first start pull request for 2.3 and second after that make modifications for 2.4, test and push again.

daschuer commented:
Ah I see. So nice the mapping has also a version in the 2.3 branch, how about rebase you changes to the 2.3 branch? This way you will see you changes soon in the stable release. I consider to  prepare a 2.3.3 after the auto DJ PRs are merged.

I have also noticed that you work on you main branch. It is recommended to create a branch for every feature. pre-commit will complain about it.

The commands are.
```
git checkout -b mc7000_23
git rebase --onto 2.3 bc835b4c4ffd3e3e718e883e9ba390339ee77a4f~1
``` 
This will stop once a conflict happens, indicated with <<<<< in the affected source.
Fix the conflicts and 
```
git commit -a --amend 
git rebase --continue
```

Give a hint if something whent wrong, than I will help.

holopansel commented:
> 

There is a merge conflict in src/library/autodj/autodjprocessor.cpp,

how do I resolve this?

daschuer commented:
> second after that make modifications for 2.4, test and push again.

That is not necessary, because we merge regularly the 2.3 branch to main. My hope is that there are only easy to solves conflict. If not I can come back to you for help.

daschuer commented:
> There is a merge conflict in src/library/autodj/autodjprocessor.cpp,

Hah, I see that has happened because your have merged from main. That can be solved by rebating this brancht to main first. 

So it is:
```
git checkout -b mc7000_23
git rebase --onto main bc835b4c4ffd3e3e718e883e9ba390339ee77a4f~1
git rebase --onto 2.3 main
```

holopansel commented:
> > There is a merge conflict in src/library/autodj/autodjprocessor.cpp,
> 
> Hah, I see that has happened because your have merged from main. That can be solved by rebating this brancht to main first.
> 
> So it is:
> 
> ```
> git checkout -b mc7000_23
> git rebase --onto main bc835b4c4ffd3e3e718e883e9ba390339ee77a4f~1
> git rebase --onto 2.3 main
> ```

Hi, this did not work, still merge conflicts.
I am on vacation currently, after my return I will start from the beginning again, create a new repository fork and start work on that directly in branch 2.3. I think this is more efficient than debugging here. Thank you for your support so far.
